src/main/java/com/jjcsa/controller/UserController.java

```
@@ -54,4 +72,29 @@ public String login(@RequestBody @NonNull final UserLogin userLogin) {
         return token;
     }
 
+    @PostMapping(path = "/register")
+    public ResponseEntity<AddNewUser> register(@RequestBody @NonNull final AddNewUser addNewUser) {
+
+        // Create the new user in keycloak
+        if(!KeycloakUtil.createNewUser(addNewUser, keycloakServerUrl)) {
```

I would suggest to add user to Keycloak only after the user is added to the database

---

I added it before to ensure if Keycloak is down, we should not create the user in our db.
Is there any other reason to do keycloak later?

---

Main reason is that if the user is not added to the DB we should not have his credentials in Keycloak as well. Since we should consider DB as the main owner of the data. So if there is any issue creating in DB we should not have that user data anywhere and should throw the exception

---

Addressed. Also added code to delete user in our db, if it failed in Keycloak.